tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TspClientTool/TspClientTool.cs

```diff
@@ -0,0 +1,476 @@
+                    ResponseError = result.Success ? null : result.ErrorMessage,
+                    Duration = result.Duration
+                };
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error initializing project with tsp-client");
+                return new DefaultCommandResponse
+                {
+                    ResponseError = $"Error initializing project: {ex.Message}"
+                };
+            }
+        }
+
+        [McpServerTool(Name = "tsp-client-update"), Description("Sync and generate from TypeSpec project")]
+        public async Task<DefaultCommandResponse> UpdateProject(
+            string? outputDir = null,
+            string? repo = null,
+            string? commit = null,
+            string? tspConfig = null,
+            string? localSpecRepo = null,
+            string? emitterOptions = null,
+            bool saveInputs = false,
+            bool skipInstall = false,
+            bool debug = false)
+        {
+            try
+            {
+                var args = new List<string> { "update" };
+
+                AddOptionalArg(args, "--output-dir", outputDir);
+                AddOptionalArg(args, "--repo", repo);
+                AddOptionalArg(args, "--commit", commit);
+                AddOptionalArg(args, "--tsp-config", tspConfig);
+                AddOptionalArg(args, "--local-spec-repo", localSpecRepo);
+                AddOptionalArg(args, "--emitter-options", emitterOptions);
+
+                if (saveInputs) args.Add("--save-inputs");
+                if (skipInstall) args.Add("--skip-install");
+                if (debug) args.Add("--debug");
+
+                var result = await ExecuteTspClientCommand(args);
+                return new DefaultCommandResponse
+                {
+                    Message = result.Success ? "Project updated successfully" : "Project update failed",
+                    ResponseError = result.Success ? null : result.ErrorMessage,
+                    Duration = result.Duration
+                };
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error updating project with tsp-client");
+                return new DefaultCommandResponse
+                {
+                    ResponseError = $"Error updating project: {ex.Message}"
+                };
+            }
+        }
+
+        [McpServerTool(Name = "tsp-client-generate"), Description("Generate client library from TypeSpec project")]
+        public async Task<DefaultCommandResponse> GenerateClientLibrary(
+            string? outputDir = null,
+            string? emitterOptions = null,
+            bool saveInputs = false,
+            bool skipInstall = false,
+            bool debug = false)
+        {
+            try
+            {
+                var args = new List<string> { "generate" };
```

Comment:
Can we all tsp init from this call if it's not initialized? This tool will have to take additional param for tspconfig path.  So  user can simply run `tsp-client-generate` with tspconfig path and tool will check if it's initialized or note and run init if required.